RoddyCore/src/de/dkfz/roddy/knowledge/files/FileObject.java

```diff
@@ -26,7 +27,7 @@ public FileObject(ExecutionContext executionContext) {
         this.executionContext = executionContext;
     }
 
-    public abstract void runDefaultOperations();
+    public abstract void runDefaultOperations() throws ConfigurationError;
```

Comment:
Why does this actually throw a configuration error? I mean it is a BaseFile class.

Also we could think of getting rid of this. The mechanism was hardly used.

Reply:
I admit that I made many of the changes, just because the compiler required them. Given the sometimes convoluted structure of Roddy, it is possible that the derived class code has the side-effect of accessing the configuration and thus the possibility to throw a ConfigurationError. If we could prevent this, this would certainly improve the code!